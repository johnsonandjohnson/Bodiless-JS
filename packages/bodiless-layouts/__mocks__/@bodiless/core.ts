const editContext = {
  activate: jest.fn(),
  isEdit: true,
};

export const useEditContext = jest.fn(() => editContext);

const activateOnEffect = {
  setId: jest.fn(),
};
export const useActivateOnEffect = () => activateOnEffect;

export const useNode = jest.fn();

const contextMenuFormInner = jest.fn();
export const contextMenuForm = jest.fn(() => contextMenuFormInner);
