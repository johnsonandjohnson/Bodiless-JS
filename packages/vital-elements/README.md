# `@bodiless/vital-elements`

For documentation on the Vital Elements package, please see:

- [Vital Elements | BodilessJS Docs](https://johnsonandjohnson.github.io/Bodiless-JS/#/VitalDesignSystem/Components/VitalElements/)
- [`doc/` Directory](./doc)
