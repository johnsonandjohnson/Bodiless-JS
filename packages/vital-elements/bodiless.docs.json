{
  "VitalDesignSystem": {
    "README.md": "./doc/VitalDesignSystem.md",
    "GettingStarted": {
      "README.md": "./doc/GettingStarted/README.md",
      "CoreConcepts.md": "./doc/GettingStarted/CoreConcepts.md",
      "ComponentsAndTokens.md": "./doc/GettingStarted/ComponentsAndTokens.md",
      "CompositionFromWithout.md": "./doc/GettingStarted/ComposingFromOutside.md",
      "ReachingInside.md": "./doc/GettingStarted/ReachingInside.md",
      "AnatomyOfA_Token.md": "./doc/GettingStarted/AnatomyOfAToken.md",
      "FunctionalClasses.md": "./doc/GettingStarted/FunctionalClasses.md",
      "WorkingWithContent.md": "./doc/GettingStarted/WorkingWithContent.md",
      "CleanComponents.md": "./doc/GettingStarted/CleanComponents.md",
      "YourFirstToken.md": "./doc/GettingStarted/YourFirstToken.md",
      "WhyVital.md": "./doc/GettingStarted/WhyVital.md"
    },
    "Components": {
      "README.md": "./doc/Components.md",
      "VitalElements": {
        "README.md": "./doc/VitalElements.md",
        "ComponentTemplate.md": "./doc/ComponentTemplate.md",
        "ExtendingAndComposingTokens.md": "./doc/ExtendingAndComposing.md",
        "SiteTypography.md": "./doc/SiteTypography.md",
        "Shadow.md": "./doc/Shadow.md",
        "StaticReplacement.md": "./doc/StaticReplacement.md",
        "Tokens": {
          "README.md": "./doc/Tokens.md",
          "TokenDomains.md": "./doc/TokenDomains.md"
        }
      },
      "assets": {
        "PXtoREMTemplate.xlsx": "./doc/assets/PXtoREMTemplate.xlsx"
      }
    }
  }
}
