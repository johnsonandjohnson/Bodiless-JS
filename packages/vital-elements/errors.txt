Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Secondary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Semantic/Color/Background/Section/Light Theme/Tertiary
 . Attempt to resolve alias to undefined
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Hover
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Border/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Light Theme/Idle
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Disabled
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Idle
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Checkbox/Background/Dark Theme/Checked
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Form Field/Text/Caption
 . Component color alias "(Typography/Body/Small" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Error
 . Component color alias "(Color/Signal/Error/Error" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Border/Success
 . Component color alias "(Color/Signal/Success/Success" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Input
 . Component color alias "(Typography/Body/Regular" is not semantic
Errors for Brand Tokens/White Label/Components/Form Field/Text/Label
 . Component color alias "(Typography/Eyebrow/Eyebrow" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Link
 . Color value "#FFFFFF" is not an alias
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Hover
 . Component color alias "(Semantic/Color/Interactive/Default" is not semantic
Errors for Brand Tokens/White Label/Components/Navigation/Primary/Primary Link/Border/Active
 . Component color alias "(Semantic/Color/Interactive/Hover" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Bottom
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Padding/Top
 . Spacing value "0" is not an alias
Errors for Brand Tokens/White Label/Components/Category Card/Background/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Category Card/Background/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Accordion/Dropdown/Border/Light Theme/Border
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Light Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Breadcrumb/Background/Dark Theme/Color
 . Component color alias "(Semantic/Color/Background/Page/Primary" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Dark Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
Errors for Brand Tokens/White Label/Components/Article Card/Border/Light Theme/Background
 . Component color alias "(Color/--NO VALUE--" is not semantic
