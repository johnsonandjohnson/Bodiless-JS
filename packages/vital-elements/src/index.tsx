/**
 * Copyright Â© 2022 Johnson & Johnson
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export * from './components/AccordionElement';
export * from './components/ArticleCardElement';
export * from './components/BreadcrumbElement';
export * from './components/ButtonElement';
export * from './components/CategoryCardElement';
export * from './components/Color';
export * from './components/DividerElement';
export * from './components/FilterElement';
export * from './components/FloatingButtonElement';
export * from './components/FontSize';
export * from './components/FooterElement';
export * from './components/HeroBannerElement';
export * from './components/ImageCarouselElement';
export * from './components/InputElement';
export * from './components/KnapsackDemoElement';
export * from './components/LinkElement';
export * from './components/ModalElement';
export * from './components/NavigationElement';
export * from './components/PDPProductOverviewElement';
export * from './components/ProductCardElement';
export * from './components/ScrollIndicatorElement';
export * from './components/Spacing';
export * from './components/TextDecoration';
export * from './components/KnapsackDemoElement';
export { default as vitalScrollIndicatorElement } from './generated/vitalScrollIndicatorElement';
export * from './components/Typography';
export * from './util';
