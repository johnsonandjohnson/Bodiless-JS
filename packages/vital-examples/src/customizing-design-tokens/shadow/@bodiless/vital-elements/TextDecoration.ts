import { shadow } from '@bodiless/vital-elements';
import { exampleTextDecoration } from '../../../components/TextDecoration';

export default shadow(exampleTextDecoration, 'Example:TextDecoration');
