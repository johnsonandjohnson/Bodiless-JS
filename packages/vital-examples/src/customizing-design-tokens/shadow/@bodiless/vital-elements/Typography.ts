import { shadow } from '@bodiless/vital-elements';
import { exampleTypography } from '../../../components/Typography';

export default shadow(exampleTypography, 'Example:Typography');
