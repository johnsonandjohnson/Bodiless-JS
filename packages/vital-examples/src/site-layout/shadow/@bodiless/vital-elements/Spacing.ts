import { shadow } from '@bodiless/vital-elements';
import { exampleSpacing } from '../../..';

export default shadow(exampleSpacing, 'Example:Spacing');
