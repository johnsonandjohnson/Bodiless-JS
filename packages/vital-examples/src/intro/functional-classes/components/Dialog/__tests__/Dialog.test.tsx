/* eslint-disable jest/valid-describe */
import React from 'react';
import { testTokens } from '@bodiless/vital-elements';
import { DialogClean, exampleDialog } from '..';

describe('Dialog Tokens', testTokens(DialogClean, exampleDialog));
