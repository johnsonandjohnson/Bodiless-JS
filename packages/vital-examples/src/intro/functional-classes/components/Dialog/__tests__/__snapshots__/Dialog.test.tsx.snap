// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Tokens Matches snapshot for Welcome 1`] = `
<div
  className="FancyBorder FancyBorder-blue"
>
  <h1
    className="Dialog-title"
  >
    Welcome
  </h1>
  <p
    className="Dialog-message"
  >
    Thank you for visiting our spacecraft!
  </p>
</div>
`;
